python/ray/util/client/server/dataservicer.py

```diff
@@ -22,6 +26,32 @@
 QUEUE_JOIN_SECONDS = 10
 
 
+def _get_reconnecting_from_context(context: Any) -> bool:
+    """
+    Get `reconnecting` from gRPC metadata, or False if not present
+    """
+    metadata = {k: v for k, v in context.invocation_metadata()}
+    val = metadata.get("reconnecting")
+    if val is None or val not in ("True", "False"):
+        logger.error(
+            f'Client connecting with invalid value for "reconnecting": {val}')
```

Comment:
when this could happen?

Reply:
Could happen if the proxier fails to forward metadata properly, but in practice this shouldn't happen and the logging is just being cautious.